nomad/fsm_test.go

```diff
@@ -3276,3 +3277,148 @@ func TestFSM_SnapshotRestore_Namespaces(t *testing.T) {
 		t.Fatalf("bad: \n%#v\n%#v", out2, ns2)
 	}
 }
+
+func TestFSM_ACLEvents_ACLToken(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		desc     string
+		setupfn  func(t *testing.T, fsm *nomadFSM)
+		raftReq  func(t *testing.T) []byte
+		reqTopic structs.Topic
+		eventfn  func(t *testing.T, e []structs.Event)
```

Comment:
I really like how this test got structured with these testing funcs as test cases. 👍

Reply:
Thanks, I think this one is right in the sweet spot for it being legible. table tests with huge tables and small test case  runs can get hard to read :man_shrugging: